lesson_7/dobricsongor/decorator.py

```diff
@@ -0,0 +1,65 @@
+# Create a decorator called uppercase that will uppercase the result
+
+def uppercase(fnc):
+    
+    def inner_func(given_text):
+        return(fnc(given_text).upper())
+    
+    return inner_func
+    
+
+@uppercase
+def greet(name):
+    return "Greetings {}!".format(name)
+
+print(greet("World"))
+
+
+#------------------------------------------------------------------------------
+
+#Create a decorator called safe_divide that will output a message if the division cannot be performed, othervise it will return the result.
+
+def safe_divide(fnc):
+
+    def inner_func(first_num,second_num):
+        if isinstance(first_num, (int,float)) and isinstance(second_num, (int,float)): 
```

Comment:
Most of the time we don't check types in Python, we simply check that certain behaviour is present on a Python object